File: python/7 kyu/406.py
# https://www.codewars.com/kata/58e0cb3634a3027180000040/train/python

# Passed

def sort_by_value_and_index(arr):
    nums_with_pos_arr = [[num, pos] for pos, num in enumerate(arr, start=1)]
    sorted_arr = list(sorted(nums_with_pos_arr, key = lambda num_pos: num_pos[0] * num_pos[1]))
    result = [num_pos[0] for num_pos in sorted_arr]
    return result

output = sort_by_value_and_index([ 23, 2, 3, 4, 5 ])
print(output)

File: nodejs/7 kyu/406.js
// https://www.codewars.com/kata/57d4e99bec16701a67000033/train/javascript

// Passed

function heavyMetalUmlauts(boringText) {
  let letterToReplacementMap = {
    A: "Ä",
    E: "Ë",
    I: "Ï",
    O: "Ö",
    U: "Ü",
    Y: "Ÿ",
    a: "ä",
    e: "ë",
    i: "ï",
    o: "ö",
    u: "ü",
    y: "ÿ",
  };

  let result = "";
  for (const char of boringText) {
    let replacement = letterToReplacementMap[char]
      ? letterToReplacementMap[char]
      : char;
    result += replacement;
  }

  return result;
}

const output = heavyMetalUmlauts("Announcing the Macbook Air Guitar");
console.log(output);